### DISCUSSION TITLE
Can I use it without Composer?

### DISCUSSION THREAD

**Discussion by mawg** [Textalk/websocket-php#174] [Q&A]
I am confused by `require __DIR__ . '/../vendor/autoload.php';`. It seems to have something to do with something called Compzer. Can I use this software without Compozer? Which `require_oce` statements would I need?

**Comment by sirn-se** [comment]
[Composer](https://getcomposer.org) is a dependency manager for PHP, having become a de facto standard in PHP community. I strongly recommend using it. Pretty much all shared PHP libraries use Composer.

But it is possible to use PHP libraries without it. First thing is to manually install all libraries it requires. And the libraries that those libraries require. And so on, until all used dependencies are resolved. You will find required dependencies in respective `composer.json` file under the `require` property.

Then you need to build your own [auto-loader](https://www.php.net/manual/en/function.spl-autoload-register.php) **or** require all involved source files in your code. For this library that means all files in `lib` directory, but you also need to figure out which files to require for installed dependencies.

Pretty complicated, yes. That's why PHP developers use Composer.
